http/middleware.go

```diff
@@ -0,0 +1,155 @@
+// Package http provides HTTP utilities and handler middleware.
+package http
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/clerk/clerk-sdk-go/v2"
+	"github.com/clerk/clerk-sdk-go/v2/jwt"
+)
+
+// RequireHeaderAuthorization will respond with HTTP 403 Forbidden if
+// the Authorization header does not contain a valid session token.
+func RequireHeaderAuthorization(ctx context.Context, opts ...AuthorizationOption) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return WithHeaderAuthorization(ctx, opts...)(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			claims, ok := clerk.SessionClaimsFromContext(r.Context())
+			if !ok || claims == nil {
+				w.WriteHeader(http.StatusForbidden)
+				return
+			}
+			next.ServeHTTP(w, r)
+		}))
+	}
+}
+
+// WithHeaderAuthorization checks the Authorization request header
+// for a valid Clerk authorization JWT. The token is parsed and verified
+// and the active session claims are written to the http.Request context.
+// The middleware uses Bearer authentication, so the Authorization header
+// is expected to have the following format:
+// Authorization: Bearer <token>
+func WithHeaderAuthorization(ctx context.Context, opts ...AuthorizationOption) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			authorization := strings.TrimSpace(r.Header.Get("Authorization"))
+			if authorization == "" {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			token := strings.TrimPrefix(authorization, "Bearer ")
+			_, err := jwt.Decode(ctx, &jwt.DecodeParams{Token: token})
```

Comment:
🔧 I think you need the Request context here:
```suggestion
			_, err := jwt.Decode(r.Context(), &jwt.DecodeParams{Token: token})
```